### DISCUSSION TITLE
Why is the Welch t-test used for ranking velocity genes when data is not normally distributed

### DISCUSSION THREAD

**Discussion by paula-tataru** [theislab/scvelo#710] [Q&A]
Hi,

According to the API, you use the Welch t-test to rank the velocity genes.
https://scvelo.readthedocs.io/scvelo.tl.rank_velocity_genes/#scvelo.tl.rank_velocity_genes

This test assumes that data is normally distributed. I got curious to see if the data meets this requirement, so I looked at the pancreas data set and tested for normality the velocity estimates for the first cluster for the top velocity gene according to the ranking.

```python
import scvelo as scv
import numpy as np
from scipy.stats import shapiro
from statsmodels.graphics.gofplots import qqplot
from matplotlib import pyplot

adata = scv.datasets.pancreas()
scv.pp.filter_and_normalize(adata, min_shared_counts=20, n_top_genes=2000)
scv.pp.moments(adata, n_neighbors=30)
scv.tl.velocity(adata, mode='deterministic')
scv.tl.rank_velocity_genes(adata, groupby='clusters')

df = scv.DataFrame(adata.uns['rank_velocity_genes']['names'])
cluster = df.columns[0]
g = np.where(adata.var_names == df[cluster][0])
p = np.where(adata.obs['clusters'] == cluster)
# velocity values for the cells in the first cluster and its top gene
v = adata.layers["velocity"][p, g][0]

pyplot.hist(v)
qqplot(v, line='s')

stat, p = shapiro(v)
print("P-value:", p)
```
Here are the two plots:
![hist](https://user-images.githubusercontent.com/4553863/134183530-0d1b52a4-943c-45d2-9f49-ce7c4d7c097b.png)
![qqplot](https://user-images.githubusercontent.com/4553863/134183555-003f0542-28bf-4679-a73a-59bc01252a05.png)

And the p-value is about 1.75e-9.

The data looks to be far from being normally distributed. Why was the Welch t-test used?

/Paula

**Comment by WeilerP** [comment]
@VolkerBergen, any comments/thoughts on this?
